java/hcx-integrator-sdk/src/main/java/io/hcxprotocol/service/NotificationService.java

```diff
@@ -0,0 +1,96 @@
+package io.hcxprotocol.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.typesafe.config.Config;
+import io.hcxprotocol.dto.NotificationRequest;
+import io.hcxprotocol.exception.ClientException;
+import io.hcxprotocol.exception.ErrorCodes;
+import io.hcxprotocol.utils.*;
+import org.apache.commons.text.StringSubstitutor;
+
+import java.util.*;
+
+import static io.hcxprotocol.utils.Constants.*;
+
+public class NotificationService {
+    private NotificationService(){
+
+    }
+    public static void validateNotificationRequest(NotificationRequest notificationRequest) throws ClientException {
+        if (notificationRequest.getTopicCode().isEmpty()) {
+            throw new ClientException("Topic code cannot be empty");
+        } else if (notificationRequest.getMessage().isEmpty() && notificationRequest.getTemplateParams().isEmpty()) {
+            throw new ClientException("Either the message or the template parameters are mandatory.");
+        } else if (notificationRequest.getRecipients().isEmpty()) {
+            throw new ClientException("Recipients cannot be empty");
+        } else if (notificationRequest.getRecipientType().isEmpty()) {
+            throw new ClientException("Recipient type cannot be empty");
+        }
+    }
+
+    private static String getMessage(NotificationRequest notificationRequest,Map<String, Object> output, String message) throws Exception {
+        Map<String, Object> searchRequest = Map.of("filters", new HashMap<>());
+        if (!notificationRequest.getTemplateParams().isEmpty()) {
+            Map<String, Object> responseMap = new HashMap<>();
+            Utils.initializeHCXCall(JSONUtils.serialize(searchRequest), Operations.NOTIFICATION_LIST, responseMap, notificationRequest.getConfig());
```

Comment:
why creating new response map and pass to initializeHCXCall? pass output map to the method.